### PR TITLE
Feature/epmdj 80

### PR DESCRIPTION
**Pull request by IgorKey**

### CODE REVIEW

File: src/adminPartMain/kotlin/CoverageAdminPart.kt [hunk 1]
```diff
@@ -340,6 +343,21 @@ class CoverageAdminPart(sender: Sender, agentInfo: AgentInfo, id: String) :
...
+        val tests = TestsToRun(
+            instanceOfStorageTestsAssociatedWithBuild.getTestsAssociatedWithMethods(
+                agentState,
+                buildMethods.allModified
+            )
+        )
+        sender.send(
+            agentInfo,
+            "/build/tests-to-run",
+            TestsToRun.serializer() stringify tests
```
Comment (IgorKey) [inline src/adminPartMain/kotlin/CoverageAdminPart.kt]: fix it. Now we are using pure object for sending. Not raw string.

File: src/adminPartMain/kotlin/Data.kt [hunk 2]
```diff
@@ -147,4 +150,9 @@ data class TestTypeSummary(
 data class Risks(
     val newMethods: List<JavaMethod>,
     val modifiedMethods: List<JavaMethod>
+)
+
+@Serializable
+data class TestsToRun(
+    val tests: List<String>?
```
Comment (IgorKey) [inline src/adminPartMain/kotlin/Data.kt]: Why is allow null here? I think this should be initialized by default with empty list

File: src/adminPartMain/kotlin/TestsAssociatedWithBuildStorage.kt [hunk 3]
```diff
@@ -0,0 +1,35 @@
+package com.epam.drill.plugins.coverage
+
+import java.util.concurrent.ConcurrentHashMap
+
+object TestsAssociatedWithBuildStorageImpl : TestsAssociatedWithBuildStorage {
```
Comment (s0larm00n) [inline src/adminPartMain/kotlin/TestsAssociatedWithBuildStorage.kt]: if I were you, would rather specify in implementation's name, which storage is used. for example MutableMapTestsAssociatedWithBuildStorage. still the interface's name is terrifying long.

File: src/adminPartMain/kotlin/TestsAssociatedWithBuildStorage.kt [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+}
+
+interface TestsAssociatedWithBuildStorage {
+    fun add(buildVersion: String, associatedTestsList: List<AssociatedTests>)
+    fun getTestsAssociatedWithMethods(agentState: AgentState, javaMethods: List<JavaMethod>): List<String>?
+    fun getStorage(): TestsAssociatedWithBuildStorage
+    fun previousBuildVersion(agentState: AgentState): String
+}
+
+val instanceOfStorageTestsAssociatedWithBuild = TestsAssociatedWithBuildStorageImpl.getStorage()
```
Comment (s0larm00n) [inline src/adminPartMain/kotlin/TestsAssociatedWithBuildStorage.kt]: no need to add 'instanceOf' prefix here. you are a kotlin programmer now, feel free to name instances by class name (in this case 'testsAssociatedWithBuildStorage')

File: src/adminPartMain/kotlin/TestsAssociatedWithBuildStorage.kt [hunk 5]
```diff
@@ -0,0 +1,35 @@
+package com.epam.drill.plugins.coverage
+
+import java.util.concurrent.ConcurrentHashMap
+
+object TestsAssociatedWithBuildStorageImpl : TestsAssociatedWithBuildStorage {
+
+    private val map: MutableMap<String, MutableSet<AssociatedTests>> = ConcurrentHashMap()
+
+    override fun getStorage() = this
```
Comment (s0larm00n) [inline src/adminPartMain/kotlin/TestsAssociatedWithBuildStorage.kt]: what is this supposed to be used for?

**Pull request merged**